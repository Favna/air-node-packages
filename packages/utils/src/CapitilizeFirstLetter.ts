/**
 * Transforms the first letter to a capital then adds all the rest after it
 * Differs from {@link sentenceCase} in that the casing of the text after the first letter is retained
 * @param str Text to transform
 */
export function capitilizeFirstLetter(str: string) {
  return str.charAt(0).toUpperCase() + str.slice(1);
}
