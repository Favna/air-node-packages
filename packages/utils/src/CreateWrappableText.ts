/**
 * Inserts a space in any text after every maxLength amount of characters, allowing Material-UI to gracefully wrap it
 * @param text text to split
 * @param maxLength The maximum length of a text section @default 75
 */
export function createWrappableText(text: string, maxLength = 75) {
  const splitRegex = new RegExp(`.{1,${maxLength}}`, 'g');
  const splitContent = text.match(splitRegex);

  if (splitContent) return splitContent.join(' ');

  return text;
}

export default createWrappableText;
