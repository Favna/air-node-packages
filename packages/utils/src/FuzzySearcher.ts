import Fuse from 'fuse.js';

export class FuzzySearcher<O> {
  private readonly object: O[];
  private readonly fuseOptions?: Fuse.IFuseOptions<unknown>;

  constructor(obj: O[], keys: string[], options?: Fuse.IFuseOptions<unknown>) {
    this.object = obj;
    this.fuseOptions = {
      threshold: 0.2,
      ...options,
      keys
    };
  }

  public runFuzzy(query: string) {
    const locquery = query.toLowerCase();

    const fuzzyFuze = new Fuse(this.object, this.fuseOptions);

    return fuzzyFuze.search<O>(locquery);
  }
}
