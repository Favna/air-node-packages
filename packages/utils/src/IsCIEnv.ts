/** Validates whether the current environment is a CI environment */
export const isCIEnv = Boolean(process.env.CI);

export default isCIEnv;