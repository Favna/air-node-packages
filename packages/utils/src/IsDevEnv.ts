/** Validates whether the current environment is either development or test */
export const isDevEnv = process.env.NODE_ENV === 'development' ? true : process.env.NODE_ENV === 'test';

export default isDevEnv;
