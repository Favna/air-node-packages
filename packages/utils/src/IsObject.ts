/**
 * Verify if the input is an object literal (or class).
 * @param input The object to verify
 */
export const isObject = (input: unknown): boolean => {
  return input ? (input as object).constructor === Object : false;
};

export default isObject;