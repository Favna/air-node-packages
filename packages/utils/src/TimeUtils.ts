export const SECOND = 1000;
export const MINUTE = SECOND * 60;
export const HOUR = MINUTE * 60;
export const DAY = HOUR * 24;

export const DAYS = [ 'Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday' ];
export const MONTHS = [ 'January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December' ];
export const DAYS_NL = [ 'Zondag', 'Maandag', 'Dinsdag', 'Woensdag', 'Donderdag', 'Vrijdag', 'Zaterdag' ];
export const MONTHS_NL = [ 'Januari', 'Februari', 'Maart', 'April', 'Mei', 'Juni', 'July', 'Augustus', 'September', 'October', 'November', 'December' ];