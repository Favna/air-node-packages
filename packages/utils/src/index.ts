export * from './IsDevEnv';
export * from './IsTestEnv';
export * from './ObjectHasProperty';
export * from './SentenceCase';
export * from './CreateWrappableText';
export * from './StableTableSort';
export * from './UseWindowSize';
export * from './CapitilizeFirstLetter';
export * from './SnakeToCamelCase';
export * from './TimeUtils';