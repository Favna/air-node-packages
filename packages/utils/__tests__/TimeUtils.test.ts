import * as TimeUtils from '../src/TimeUtils';

test('Seconds should be 1000 milliseconds', () => {
  expect(TimeUtils.SECOND).toBe(1000);
});

test('Minutes should be 60 seconds', () => {
  expect(TimeUtils.MINUTE).toBe(TimeUtils.SECOND * 60);
  expect(TimeUtils.MINUTE).toBe(60 * 1000);
});

test('Hours should be 60 minutes', () => {
  expect(TimeUtils.HOUR).toBe(TimeUtils.MINUTE * 60);
  expect(TimeUtils.HOUR).toBe(TimeUtils.SECOND * 60 * 60);
  expect(TimeUtils.HOUR).toBe(60 * 60 * 1000);
});

test('Day should be 24 hours', () => {
  expect(TimeUtils.DAY).toBe(TimeUtils.HOUR * 24);
  expect(TimeUtils.DAY).toBe(TimeUtils.MINUTE * 60 * 24);
  expect(TimeUtils.DAY).toBe(TimeUtils.SECOND * 60 * 60 * 24);
  expect(TimeUtils.DAY).toBe(24 * 60 * 60 * 1000);
});

test('DAYS should be English days of the week', () => {
  expect(TimeUtils.DAYS).toStrictEqual(['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday']);
});

test('MONTHS should be English months of the year', () => {
  expect(TimeUtils.MONTHS).toStrictEqual([
    'January',
    'February',
    'March',
    'April',
    'May',
    'June',
    'July',
    'August',
    'September',
    'October',
    'November',
    'December'
  ]);
});

test('DAYS_NL should be Dutch days of the week', () => {
  expect(TimeUtils.DAYS_NL).toStrictEqual([
    'Zondag',
    'Maandag',
    'Dinsdag',
    'Woensdag',
    'Donderdag',
    'Vrijdag',
    'Zaterdag'
  ]);
});

test('MONTHS_NL should be Dutch months of the year', () => {
  expect(TimeUtils.MONTHS_NL).toStrictEqual([
    'Januari',
    'Februari',
    'Maart',
    'April',
    'Mei',
    'Juni',
    'July',
    'Augustus',
    'September',
    'October',
    'November',
    'December'
  ]);
});

test('TimeUtils should match snapshot', () => {
  expect(TimeUtils).toMatchSnapshot();
});
