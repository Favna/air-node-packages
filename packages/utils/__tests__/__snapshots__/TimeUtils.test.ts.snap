// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeUtils should match snapshot 1`] = `
Object {
  "DAY": 86400000,
  "DAYS": Array [
    "Sunday",
    "Monday",
    "Tuesday",
    "Wednesday",
    "Thursday",
    "Friday",
    "Saturday",
  ],
  "DAYS_NL": Array [
    "Zondag",
    "Maandag",
    "Dinsdag",
    "Woensdag",
    "Donderdag",
    "Vrijdag",
    "Zaterdag",
  ],
  "HOUR": 3600000,
  "MINUTE": 60000,
  "MONTHS": Array [
    "January",
    "February",
    "March",
    "April",
    "May",
    "June",
    "July",
    "August",
    "September",
    "October",
    "November",
    "December",
  ],
  "MONTHS_NL": Array [
    "Januari",
    "Februari",
    "Maart",
    "April",
    "Mei",
    "Juni",
    "July",
    "Augustus",
    "September",
    "October",
    "November",
    "December",
  ],
  "SECOND": 1000,
}
`;
